yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -219,10 +219,25 @@ def probe_executable(self):
 
     @staticmethod
     def stream_copy_opts(copy=True, *, ext=None):
-        yield from ('-map', '0')
+        if ext in ('mkv', 'mka'):
+            yield from ('-map', '0')
+
+            # Some streams, such as JSON attachments, are considered of unknown
+            # type by FFmpeg but we still want to copy them.
+            yield '-copy_unknown'
+        else:
+            # Other containers can get angry at malformed or unknown streams,
+            # so we'll keep only the ones that are "usable".
+            yield from ('-map', '0:u')
```

Comment:
>  - otherwise, we'll pass `-ignore_unknown` just in case and a different map operation, `-map 0:u`. The `u` stream specifier, according to FFmpeg man page, selects only "usable" streams, with at least a codec and whatever related variable it needs to have to be called as such. This should make sure that we only select properly formed streams which please the pickier, non-generalist, containers.

Since this PR really has nothing to do with non-matroska files, this change doesn't belong in this PR IMO

Reply:
wdym broken remuxing? this PR shouldn't be changing anything for non-matroska media, right?

Reply:
I liked the `if ext in ('mkv', 'mka'):` code path, what I didn't like was changing the behavior for the `else` path to use `0:u` instead of `0`

Reply:
yep that'd be good